src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1624,7 +1624,17 @@ public function touchOwners()
 
 			if ( ! is_null($this->$relation))
```

Comment:
Given `instanceof` checks, `is_null` is redundant. This will do:
```php
foreach ($this->touches as $relation)
{
	$this->$relation()->touch();

	if ($this->$relation instanceof Model)
	{
		$this->$relation->touchOwners();
	}
	elseif ($this->$relation instanceof Collection)
	{
		$this->$relation->each(function (Model $related)
		{
			$related->touch();
		});
	}
}
```


And btw I agree that it belongs to the `Eloquent\Collection`, but as far as I know @taylorotwell didn't want it there?